docs/source/installation.rst

```diff
@@ -100,98 +100,270 @@ Step 1: Setup dedicated python environment
             If you are using `Compute Canada <https://www.computecanada.ca/>`_, you can load modules as `mentioned here <https://intranet.neuro.polymtl.ca/computing-resources/compute-canada#modules>`_ and `also here <https://docs.computecanada.ca/wiki/Utiliser_des_modules/en#Loading_modules_automatically>`_.
 
 
-Step 2: Install ``ivadomed``
-++++++++++++++++++++++++++++
-
+Step 2: Install ``ivadomed`` with GPU or CPU Support
+++++++++++++++++++++++++++++++++++++++++++++++++++++
 
     .. tabs::
+        
+        .. tab:: NVIDIA GPU Support
+
+            ``ivadomed`` requires a minimum PyTorch version of either
+            1.8.1 or 1.8.2 which supports CUDA 10.2 and CUDA 11.1 builds
+            by default. 
+            
+            Ampere-based GPUs (with a Compute Capability of 8.x) natively
+            support CUDA>=11.1. Although CUDA 11.1 is backward compatible
+            but CUDA 10.2 is preferred for older GPUs.
+
+            CUDA 10.2 and CUDA 11.1 require an NVIDIA driver version >=440.33 
+            and >=450 respectively as indicated `here <https://docs.nvidia.com/deploy/cuda-compatibility/index.html#minor-version-compatibility>`__.
+            Please make sure to upgrade to the minimum NVIDIA driver version 
+            requirements for the respective CUDA builds. To verify the NVIDIA
+            driver version, just run the command `nvidia-smi` and you'll find 
+            your current driver version on the top.
```

Comment:
There's also the `pip.conf` solution: instead of "please run this command" it could be "please create this file before installation"


```
# /etc/pip.conf or ~/.config/pip/pip.conf or C:\ProgramData\pip\pip.ini or %APPDATA%\pip\pip.ini
[install]
extra-index-url =
   https://download.pytorch.org/whl/cu111
   https://download.pytorch.org/whl/cu113
   https://download.pytorch.org/whl/cu115
```

That's essentially instructing people to set a global variable on their OS, but I think this is one of the rare cases where that's the right thing to do: a system has fixed hardware (usually), so every pretty much every package they use with their GPUs will need the same patch.

Reply:
> I also wonder if even mentioning "10.2" will mislead people. 

Maybe or maybe not

> It's not really CUDA 10.2 that's important, but rather just using the default `torch` that's on PyPI, because that's the best-tested (and evidently most optimized) version of `torch` out there. Similarly the "minimum version" of torch is encoded in the .whl, so that's also something users shouldn't need to think about. So maybe:

Agreed

> (I replaced `nvidia-smi` with `cat` because `nvidia-smi` isn't installed with the drivers; on Ubuntu it comes [with `nvidia-cuda-toolkit`](https://github.com/neuropoly/computers/blob/99c8bf5c24365f7eb1f71d252011f605ead425c7/ansible/roles/neuropoly.gpu/tasks/main.yml#L19) but not with `nvidia-driver-XXX`; 

Thank you for adding this... I was sceptical about this as well

> I suppose for someone who was trying to get things working on Windows, this is a step back, since there maybe `nvidia-smi` does come? But on Windows someone can probably look in Control Panel to find the driver version?)
Yuss yuss

> `ivadomed-test-gpus` is a new suggestion, suggesting we make sure people can actually. We both know it could be `python -c 'import torch; print(torch.cuda.is_available())`, but maybe we should wrap it in a small helper script to prove to people that ivadomed is actually talking to the GPUs? Notice that except for "download.pytorch.org" we don't need to mention torch at all -- it's just a dependency, it's not what people are looking to learn about from ivadomed so the more we avoid opening the rabbit hole the better I think. But if we did ship such a script we'd have to maitnain it; so is there an obvious command that already exists in ivadomed that reports whether it's using GPUs or not that people could run on some small test data, just to verify everything?
I agree... I'm thinking something along the lines what ADS has ([Quick Test](https://axondeepseg.readthedocs.io/en/latest/documentation.html#quick-test)). It should come after the installation step... let's skip this for now

Reply:
> There's also the `pip.conf` solution: instead of "please run this command" it could be "please create this file before installation"
> 
> ```
> # /etc/pip.conf or ~/.config/pip/pip.conf or C:\ProgramData\pip\pip.ini or %APPDATA%\pip\pip.ini
> [install]
> extra-index-url =
>    https://download.pytorch.org/whl/cu111
>    https://download.pytorch.org/whl/cu113
>    https://download.pytorch.org/whl/cu115
> ```
> 
> That's essentially instructing people to set a global variable on their OS, but I think this is one of the rare cases where that's the right thing to do: a system has fixed hardware (usually), so every pretty much every package they use with their GPUs will need the same patch.

I'm a little hesitant towards this solution bcz let's say someone follows these instructions and later on forgets this setting, and perhaps the torch installation evolves, then it'd be harder for them to debug.